File: pow/pow.py
class Solution:
    # @param x, a float
    # @param n, a integer
    # @return a float
    def pow(self, x, n):
        returnnum = 1.0
        if n == 0:
            return 1
        elif n > 0:
            for i in range(0,n):
                returnnum = returnnum *x
#            return x*self.pow(x,n-1)
        elif n < 0:
            for i in range(n,0):
                returnnum = returnnum/x
#            return 1/x*self.pow(x,n+1)
        return returnnum
        
s = Solution()
print s.pow(0.00001,2147483647)

File: pow/pow.cpp
//what a boring question and a boring solution! no use at all


class Solution {
public:
    double pow(double x, int n) {
        int i,j,k;
        double returnnum = 1;
        if(n==0)
            return 1;
        else if(n>0)
        {
            while(n>0)
            {

                if(returnnum*x*x == returnnum)
                {
                    if (n%2==1)
                        return returnnum*x;
                    else
                        return returnnum;
                }
                returnnum*=x;
                n--;
            }
        }
        else if ( n< 0)
        {
            while(n<0)
            {

                if(returnnum/x/x == returnnum)
                {
                    if (n%2==1)
                        return returnnum/x;
                    else
                        return returnnum;               
                }
                returnnum = returnnum / x;
                n++;
            }
        }
        return returnnum;
    }
};